lib/mrb/scripts/command_line/grndb.rb

```diff
@@ -423,12 +427,20 @@ def recover
         end
 
         def truncate_corrupt_objects
+          @database.clear_lock if @database.locked?
           @database.each do |object|
-            next unless object.corrupt?
+            unless object.corrupt?
+              case object
+              when Table, Column
+                object.clear_lock if object.locked?
+              end
+              next
+            end
             logger = @context.logger
             object_path = object.path
             object_dirname = File.dirname(object_path)
             object_basename = File.basename(object_path)
+            object.clear_lock if object.locked?
```

Comment:
At first, I've misunderstood behavior of truncated object. No need to clear_lock.